Code/GraphMol/DistGeomHelpers/BoundsMatrixBuilder.cpp

```diff
@@ -868,36 +868,33 @@ bool _checkMacrocycleAllInSameRingAmideEster14(const ROMol &mol, const Bond *,
 
   if (a3Num != 6) return false;
 
-  ROMol::ADJ_ITER nbrIdx, endNbrs;
   if (a2Num == 7 || a2Num == 8) {
     if (mol.getAtomDegree(atm2) == 3 && mol.getAtomDegree(atm3) == 3) {
-      boost::tie(nbrIdx, endNbrs) = mol.getAtomNeighbors(atm2);
-      while (nbrIdx != endNbrs) {
-        if (*nbrIdx != atm1->getIdx() && *nbrIdx != atm3->getIdx()) {
-          const auto &res = mol.getAtomWithIdx(*nbrIdx);
-          const auto &resbnd = mol.getBondBetweenAtoms(atm2->getIdx(), *nbrIdx);
+      for (auto nbrIdx :
+           boost::make_iterator_range(mol.getAtomNeighbors(atm2))) {
```

Comment:
Just `mol.atomNeighbors()` ?

Reply:
We need both the atom index and the atom itself here, so either one works. Since the atom indices get used first and potentially make it so that the atom itself never needs to be retrieved, I went with that.